v2/pkg/tcr/connectionpool.go

```diff
@@ -196,16 +211,25 @@ func (cp *ConnectionPool) ReturnConnection(connHost *ConnectionHost, flag bool)
 		cp.flagConnection(connHost.ConnectionID)
 	}
 
+	// upon shutdown this would return an error.
+	// at this point the (e.g. http request) processing would have
+	// already finished, so the user does not need to handle this error
 	_ = cp.connections.Put(connHost)
```

Comment:
in case the queue has been disposed of, it makes sense to properly close the "returned"/passed connHost